packages/core/src/extension/extension-messenger.ts

```diff
@@ -94,22 +110,11 @@ export class ExtensionMessenger {
         return extension;
     }
 
-    private _validateCommand<T extends keyof ExtensionCommandMap>(command: string): T {
-        if (this._isExtensionCommandType<T>(command)) {
+    private _validateCommand<T extends keyof ExtensionCommandMap>(command: T): T {
+        if (Object.values(ExtensionCommandType).includes(command)) {
```

Comment:
@bc-peng I can't remember why we had the generic typeguard here before? This change looks good, can you please check as well.

Reply:
We need a typeguard here because of `const`, and DC removed it in this PR.
<img width="500" alt="image" src="https://github.com/bigcommerce/checkout-sdk-js/assets/88361607/64cc89ba-a8ad-4d3c-91c9-0808b5887ab5">

Reply:
Perfect 👏

Reply:
@animesh1987 I'm also exporting `ExtensionCommandType` explicitly so it can be imported as a regular object. Previously, we're only exporting it as a TS enum (`const enum`) so clients can't use it as a value.